src/cpp/src/whisper_pipeline.cpp

```diff
@@ -41,21 +45,17 @@ std::vector<int64_t> whisper_generate(const ov::genai::WhisperGenerationConfig&
                                       ov::genai::WhisperFeatureExtractor& feature_extractor,
                                       const std::shared_ptr<StreamerBase> streamer);
 
-class WhisperPipeline::Impl {
+class WhisperPipelineImpl : public WhisperPipelineImplBase {
```

Comment:
```suggestion
class WhisperPipelineStatefulImpl : public WhisperPipelineImplBase {
```

And I would separate `WhisperPipelineStatefulImpl` and `WhisperPipeline` classes to different source files.